File: backend/models/Product.js
const mongoose = require('mongoose');

const Schema = mongoose.Schema; 

const productSchema = new Schema({
    code:{type:String,required:true},
    name:{type:String,required:true},
    price:{type:Number,required:true},
    description:{type:String,required:true},
    smallDescription:{type:String,required:true},
    image:{type:String,required:true},
    category:{type:String,required:true},
    brand:{type:String,required:true},
    countInStock:{type:Number,required:true}, 
},{timestamps:true});

module.exports = mongoose.model('Product',productSchema);

File: frontend/lib/models/product.dart
class Product {
  final String code;
  final String name;
  final String description;
  final String smallDescription;
  final int price;
  final String imageUrl;
  final String category;


  Product({
    required this.code,
    required this.name,
    required this.description,
    required this.smallDescription,
    required this.price,
    required this.imageUrl,
    required this.category,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      code: json['code'],
      name: json['name'],
      description: json['description'],
      smallDescription: json['smallDescription'],
      price: json['price'],
      imageUrl: json['imageUrl'],
      category: json['category'],
    );
  }

  

}